methods/array_ops/src/pg_gp/array_ops.c

```diff
@@ -702,8 +702,14 @@ array_sub(PG_FUNCTION_ARGS){
 PG_FUNCTION_INFO_V1(array_mult);
 Datum
 array_mult(PG_FUNCTION_ARGS){
-    if (PG_ARGISNULL(0)) { PG_RETURN_NULL(); }
-    if (PG_ARGISNULL(1)) { PG_RETURN_NULL(); }
+    if (PG_ARGISNULL(0) && PG_ARGISNULL(1)) { PG_RETURN_NULL(); }
```

Comment:
Generally I try to avoid treating NULL as `zero` in multiplication because it produces zero even though only one entry of your data is NULL. Having said that, in numpy `np.nan*2 = np.nan`. But  `np.nan + 2` also results in np.nan.